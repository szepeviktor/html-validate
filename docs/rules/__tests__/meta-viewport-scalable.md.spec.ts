import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1" />`;
markup["correct"] = `<!-- explicily enabled -->
<meta name="viewport" content="user-scalable=yes" />

<!-- property no set (default enabled) -->
<meta name="viewport" content="width=device-width, initial-scale=1" />`;

describe("docs/rules/meta-viewport-scalable.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"meta-viewport-scalable":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"meta-viewport-scalable":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
});
