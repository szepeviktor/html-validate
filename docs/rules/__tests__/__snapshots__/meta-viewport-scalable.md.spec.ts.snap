// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/meta-viewport-scalable.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/meta-viewport-scalable.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 52,
        "context": undefined,
        "line": 1,
        "message": "\\"user-scalable\\" must not be disabled",
        "offset": 51,
        "ruleId": "meta-viewport-scalable",
        "ruleUrl": "https://html-validate.org/rules/meta-viewport-scalable.html",
        "selector": "meta",
        "severity": 2,
        "size": 16,
      },
    ],
    "source": "<meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, initial-scale=1\\" />",
    "warningCount": 0,
  },
]
`;
